app/src/main/java/com/github/mik629/android/fundamentals/data/background/UpdateCacheWorker.kt

```diff
@@ -0,0 +1,43 @@
+package com.github.mik629.android.fundamentals.data.background
+
+import android.content.Context
+import androidx.work.CoroutineWorker
+import androidx.work.ListenableWorker
+import androidx.work.WorkerFactory
+import androidx.work.WorkerParameters
+import com.github.mik629.android.fundamentals.domain.repositories.MoviesRepository
+import timber.log.Timber
+import javax.inject.Inject
+
+class UpdateCacheWorker(
+    context: Context,
+    workerParameters: WorkerParameters,
+    private val moviesRepository: MoviesRepository
+) : CoroutineWorker(context, workerParameters) {
+
+    override suspend fun doWork(): Result =
+        try {
+            Timber.d("Running background task for cache update")
+            moviesRepository.loadMoviesFromNetwork(category = "upcoming")
```

Comment:
maybe move category inside repo implementation if it is static

Reply:
No, it's not static, I used it to load different movies in worker (so that I could see the result of the worker)